x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/TlsRuntimeCheck.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.support;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xpack.core.XPackSettings;
+
+import java.util.function.Consumer;
+
+public class TlsRuntimeCheck {
+
+    private static final Logger logger = LogManager.getLogger(TlsRuntimeCheck.class);
+
+    private final Settings settings;
+    private final boolean httpTlsEnabled;
+    private final boolean transportTlsEnabled;
+
+    public TlsRuntimeCheck(Settings settings) {
+        this.settings = settings;
+        this.httpTlsEnabled = XPackSettings.HTTP_SSL_ENABLED.get(settings);
+        this.transportTlsEnabled = XPackSettings.TRANSPORT_SSL_ENABLED.get(settings);
```

Comment:
I don't think we want to check TransportTLS here.

1. It requires transport TLS even when in dev mode
2. It means you need to set `transport.ssl.enabled` even if using single node discovery.
3. It duplicates checks that are done elsewhere

Since security already has checks for transport TLS (outside of trial / dev mode) I don't think it's worth duplicating them here.

Reply:
Makes sense I'll remove it and make it only check the HTTP interface.

Reply:
When you say "security already has checks for transport TLS", do you mean [TLSLicenseBootstrapCheck](https://github.com/elastic/elasticsearch/blob/7.12/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/TLSLicenseBootstrapCheck.java) which errors out if the cluster is in non-trial production? Also I assume the transport TLS check in [TransportNodesReloadSecureSettingsAction](https://github.com/elastic/elasticsearch/blob/7.12/server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java#L77) is redundant as well?